### PR TITLE
 API interaction under handle data is updated

### PR DESCRIPTION
**Pull request by sibteali786**

### CODE REVIEW

File: build/handle-data/item-template/1-interaction-with-api.md [hunk 1]
```diff
@@ -1,39 +1,227 @@
-# Basic Example of Data from API
+# Api
 
-**item-template** is a set of attributes that enable the 'quick' creation of an Item. It is used to display the data from an API.
+API widget with documented API not only saves development time but also ensures consistency across and improves the user experience of the app.
```
Comment (amin-nas) [inline build/handle-data/item-template/1-interaction-with-api.md]: API is not a widget. That aside, this sentence doesn't make sense. How does it save development time or improve UX?

File: build/handle-data/item-template/1-interaction-with-api.md [hunk 2]
```diff
@@ -1,39 +1,227 @@
...
-**name:** Set the name to reference as you iterate through the array of data
+## Set Values
 
-**template:** The widget to render for each item
+First of all we have to set different values in the API widget before we move to actually using the response from the API itself.
 
-## Example
+#### Specify Input
+
+Let specify our inputs for the API if there are any inputs value we want to provide to the api uri endpoint. For now we do not have any input but let us see if there how would deal with it.
```
Comment (amin-nas) [inline build/handle-data/item-template/1-interaction-with-api.md]: Sentence doesn't seem to be proper English.

File: build/handle-data/item-template/1-interaction-with-api.md [hunk 3]
```diff
@@ -1,39 +1,227 @@
...
+    records:
+      - fields:
+          desc: "${name}"
+```
+
+</div>
+
+### Best Practices
+
+Some properties are always required and minimum ones required to make an API request work in actual.
```
Comment (amin-nas) [inline build/handle-data/item-template/1-interaction-with-api.md]: We should mark required props in the reference table. Then we dont need this.

File: build/handle-data/item-template/1-interaction-with-api.md (hunk 2 above)
Comment (sibteali786) [reply build/handle-data/item-template/1-interaction-with-api.md]: let me update it have just saw now and its grammar is totally broken :))

**Review by amin-nas — APPROVED** [review]

**Pull request merged**